sdk/storage/azure_storage_blob/src/models/blob_properties.rs

```diff
@@ -62,38 +63,38 @@ impl FromHeaders for BlobProperties {
             .unwrap();
         properties.access_tier_inferred = Some(access_tier_inferred);
 
-        let access_tier: AccessTier = headers.get_optional_as(&BLOB_ACCESS_TIER).unwrap().unwrap();
-        properties.access_tier = Some(access_tier);
+        let access_tier: Option<AccessTier> = headers.get_optional_as(&BLOB_ACCESS_TIER).unwrap();
```

Comment:
Never `unwrap()` in production code. That will panic and tear the whole process down. You already return a `Result` from this, so you use the `?` operator e.g.,
```suggestion
        let access_tier: Option<AccessTier> = headers.get_optional_as(&BLOB_ACCESS_TIER)?;
```
If you get a compiler error that there's no `From<T>` for that conversion, please let me know. For one, `from_headers` should be returning an `azure_core::Result<Option<Self>>` (we default to `azure_core::Error`) and we should already have a bunch of conversions anyway.

So no `unwrap()` or `expect()` in production code. Tests are fine. Panicking fails the test, which is probably what you want if unexpected anyway. Things like `unwrap_or_default()` are okay when used correctly, like if you need an empty string for a `String` if nothing is specified.

Reply:
Hi @heaths Heath, I believe you responded to an old commit here, since this has been resolved in the latest revision:
```rust
let access_tier: Option<AccessTier> = headers.get_optional_as(&BLOB_ACCESS_TIER)?;
```

Reply:
Not sure what happened. I just clicked on the View updates button in GH.